### PR TITLE
Token Icons - Add missing icons in UI - (excluding YB tokens)

### PR DESCRIPTION
**Pull request by orenku**
This still doesn't include the AuraBal - I will fix it in another PR

### CODE REVIEW

File: src/components/AssetLogo.tsx [hunk 1]
```diff
@@ -1,26 +1,47 @@
...
+}) => {
+  const { logoURI } = useCurrencyLogoURI(tokenAddress || "0x")
+
+  const getImage = (name: string) => {
+    switch (name) {
+      case "curve":
+        return <Image src={CurveLogo} priority alt="" />
+      case "balancer":
+        return <Image src={BalancerLogo} priority alt="" />
+      case "crypto":
```
Comment (vanillaHill) [inline src/components/AssetLogo.tsx:31]: We have only aurabal in crypto now but it won't be the case in the future. It's not a better way to handle the crypto category even if it's working currently imo.

File: src/components/AssetLogo.tsx [hunk 2]
```diff
@@ -1,26 +1,47 @@
...
+        return <Image src={AuraLogo} priority alt="" />
+      case "token":
```
Comment (vanillaHill) [inline src/components/AssetLogo.tsx:33]: Normally token no longer exists, replaced by featured/crypto/stable

File: src/components/MultiLayerTokenLogo.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  return (
+    <span id="mull" className="relative h-8 w-8" style={{ fontSize: 12 }}>
+      <BaseLogo
+        vaultType={vaultType}
+        isLpToken={isLpToken}
+        className={className}
+      ></BaseLogo>
+      {tokens &&
+        tokens.map((token, index) => {
+          const top: string = size / 1.5 - 1 + positions[index].x + "px"
```
Comment (vanillaHill) [inline src/components/MultiLayerTokenLogo.tsx:93]: You should use variables for these magic numbers, thus the code is easier to read.

**Review by vanillaHill — CHANGES_REQUESTED** [review]

File: src/components/MultiLayerTokenLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+  const positions = getSubLogosPositions(tokens?.length, size / 4, size / 4)
+
+  return (
+    <span id="mull" className="relative h-8 w-8" style={{ fontSize: 12 }}>
+      <BaseLogo
+        vaultType={vaultType}
+        isLpToken={isLpToken}
+        className={className}
+      ></BaseLogo>
```
Comment (aquaticone) [inline src/components/MultiLayerTokenLogo.tsx:90]: It's cleaner to self-close tags

```tsx
<BaseLogo ... />
```

File: src/components/MultiLayerTokenLogo.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      _positions.push({ x: posX, y: posY })
+    }
+
+    return _positions
+  }
+
+  const positions = getSubLogosPositions(tokens?.length, size / 4, size / 4)
+
+  return (
+    <span id="mull" className="relative h-8 w-8" style={{ fontSize: 12 }}>
```
Comment (aquaticone) [inline src/components/MultiLayerTokenLogo.tsx:85]: Remove the `id`

File: src/components/TokenForm/TokenSelectModal.tsx [hunk 6]
```diff
@@ -66,10 +74,24 @@ const TokenSelectModal: FC<TokenSelectModalProps> = ({
                   onClick={clickHandler}
                   onKeyDown={keyHandler}
                 >
-                  <AssetLogo
-                    className="col-start-1 row-span-2 row-start-1 h-7 w-7"
-                    name="curve"
-                  />
+                  {" "}
```
Comment (aquaticone) [inline src/components/TokenForm/TokenSelectModal.tsx:77]: Is this needed?

File: src/components/Vault/VaultDepositForm.tsx [hunk 7]
```diff
@@ -209,9 +209,10 @@ const VaultDepositForm: FC<VaultProps> = (props) => {
           onSubmit={onSubmitForm}
           submitText={requiresApproval ? "Approve" : "Deposit"}
           tokenAddresses={[
-            ...(lpTokenOrAsset ? [lpTokenOrAsset ?? "0x"] : []),
```
Comment (aquaticone) [inline src/components/Vault/VaultDepositForm.tsx:212]: Should this be removed?

File: src/hooks/useTokensOrNative.ts [hunk 8]
```diff
@@ -14,15 +14,19 @@ import useActiveChainId from "@/hooks/useActiveChainId"
...
 }
 
 export default function useTokensOrNative({
-  addresses = ["0x"],
+  tokenAddresses = ["0x"],
+  lpToken,
 }: {
-  addresses: Address[] | readonly Address[] | undefined
+  tokenAddresses: Address[] | readonly Address[] | undefined
+  lpToken: Address | Address | undefined
```
Comment (aquaticone) [inline src/hooks/useTokensOrNative.ts:25]: `Address` is here twice

**Review by aquaticone — APPROVED** [review]

**Review by aquaticone — CHANGES_REQUESTED** [review]
Pressed approve by accident

File: next.config.js [hunk 9]
```diff
@@ -23,6 +23,16 @@ const nextConfig = {
 
     return config
   },
+  images: {
+    remotePatterns: [
+      {
+        protocol: "https",
+        hostname: "raw.githubusercontent.com",
+        port: "",
+        pathname: "/FortressFinance/assets/master/blockchains/**",
```
[the remark this answers is not in the document]
Comment (orenku) [reply next.config.js:32]: I'll create a task to be handled in the future

File: src/components/AssetLogo.tsx [hunk 10]
```diff
@@ -1,26 +1,47 @@
...
+
+import AuraLogo from "~/images/assets/aura.png"
 import BalancerLogo from "~/images/assets/balancer.png"
 import CurveLogo from "~/images/assets/curve.png"
 
 type AssetLogoProps = {
   name: string
   className: string
+  style?: CSSProperties
+  tokenAddress?: `0x${string}`
```
[the remark this answers is not in the document]
Comment (orenku) [reply src/components/AssetLogo.tsx:14]: Fixed

File: src/hooks/useTokensOrNative.ts (hunk 8 above)
Comment (orenku) [reply src/hooks/useTokensOrNative.ts:25]: removed

File: src/components/MultiLayerTokenLogo.tsx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+  isLpToken,
+  className,
+  size,
+}) => {
+  const getSubLogosPositions = (
+    numberOfCurrencyIcons: number | undefined,
+    rx: number,
+    ry: number
+  ) => {
+    const _positions: { x: number; y: number }[] = []
```
[the remark this answers is not in the document]
Comment (orenku) [reply src/components/MultiLayerTokenLogo.tsx:63]: Just internal variable naming - to differentiate it from the **const positions**  used in the component itself

File: src/components/AssetLogo.tsx [hunk 12]
```diff
@@ -1,26 +1,47 @@
...
+  name,
+  tokenAddress,
+  style,
+}) => {
+  const { logoURI } = useCurrencyLogoURI(tokenAddress || "0x")
+
+  const getImage = (name: string) => {
+    switch (name) {
+      case "curve":
+        return <Image src={CurveLogo} priority alt="" />
```
[the remark this answers is not in the document]
Comment (orenku) [reply src/components/AssetLogo.tsx:28]: That repo is a mapping between addresses (token/contract)  to logo
It is not used for general purpose logos such as Curve, Balancer or even Fortress logo

File: src/components/TokenForm/TokenSelectModal.tsx (hunk 6 above)
Comment (orenku) [reply src/components/TokenForm/TokenSelectModal.tsx:77]: No :-)

**Review by aquaticone — APPROVED** [review]

**Pull request merged**